src/app/utilities/AnnotateUtilities.ts

```diff
@@ -2,17 +2,70 @@
 import jp from 'jsonpath';
 
 /* Import Utilities */
-import { ExtractFromSchema, ExtractClassesAndTermsFromSchema } from 'app/utilities/SchemaUtilities';
+import { ExtractFromSchema, ExtractClassesAndTermsFromSchema, MakeJsonPathReadableString } from 'app/utilities/SchemaUtilities';
 
 /* Import Types */
 import { DigitalSpecimen } from "app/types/DigitalSpecimen";
 import { DigitalMedia } from "app/types/DigitalMedia";
-import { AnnotationFormProperty, Dict } from "app/Types";
+import { AnnotationFormProperty, AnnotationTarget, ParentClass, Dict } from "app/Types";
 
 
 /* Utilities associated with annotating */
 
 
+/**
+ * Function to extract parent classes from a given super class and provide all their associated information
+ * @param params An object containing: the annotationTarget, superClass and parentClasses properties
+ * @returns The extracted parent classes
+ */
+const ExtractParentClasses = (params: {
+    annotationTarget: AnnotationTarget,
+    superClass: DigitalSpecimen | DigitalMedia | Dict
+}) => {
+    const { annotationTarget, superClass } = params;
+
+    /* Base variables */
+    const pathArray: string[] = ['$'];
+    const parents = jp.parse(annotationTarget.jsonPath).slice(1, -1);
+    const parentClasses: ParentClass[] = [];
+
+    /* For each parant, find and extract its class information */
+    parents.forEach((parent, index) => {
+        /* Filter path from schemas */
+        let parentName = parent.expression.value.replace('ods:', '').replace('dwc:', '').replace('dcterms:', '');
```

Comment:
Handy in `MakeJsonPathReadableString` too

Reply:
Implemented it for `ExtractParents` function, `MakeJsonPathReadableString` seems not to work since there can be multiple of the same ods:, dwc: or dcterms:

Reply:
Good call, added those in the latest PR!